File: Example/DeepleaperAdExample/DeepleaperAdExample/WebViewController.m
//
//  WebViewController.m
//  DeepleaperAdExample
//
//  Created by 刘畅 on 2017/7/10.
//  Copyright © 2017年 刘畅. All rights reserved.
//

#import "WebViewController.h"
#import "FakeData.h"
#import "LandPageViewController.h"
@import DeepleaperAd;

@interface WebViewController ()<DeepleaperFeedAdManagerDelegate>
@property (weak, nonatomic) IBOutlet UITableView *tableView;
@property(nonatomic,copy) NSMutableArray* sourceArray;
@property(nonatomic,assign) NSInteger indexRow;
@property(nonatomic,strong) DeepleaperFeedAdManager* fam;
@end

@implementation WebViewController

-(NSMutableArray*)sourceArray{
    if (!_sourceArray) {
        _sourceArray = [NSMutableArray new];
    }
    return _sourceArray;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    for (NSUInteger i = 0 ; i<50; i++) {
        FakeData* data = [FakeData new];
        data.title = [NSString stringWithFormat:@"第%ld行",(unsigned long)i];
        [self.sourceArray addObject:data];
    }
}
- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath{
    return UITableViewAutomaticDimension;
}

- (CGFloat)tableView:(UITableView *)tableView estimatedHeightForRowAtIndexPath:(NSIndexPath *)indexPath{
    return UITableViewAutomaticDimension;
}
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return _sourceArray.count;
}
- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    tableView.separatorStyle = UITableViewCellSeparatorStyleSingleLine;
    FakeData* data = _sourceArray[indexPath.row];
    if (data.title) {
        static NSString *ID = @"cell";
        UITableViewCell* cell = [tableView dequeueReusableCellWithIdentifier:ID];
        if (cell == nil) {
            cell = [[UITableViewCell alloc]initWithStyle:UITableViewCellStyleDefault reuseIdentifier:ID];
        }
        cell.textLabel.text = data.title;
        return cell;
    }
    //广告
    else{
        static NSString *adID = @"adCell";
        UITableViewCell* cell = [tableView dequeueReusableCellWithIdentifier:adID];
        if (cell == nil) {
            cell = [[UITableViewCell alloc]initWithStyle:UITableViewCellStyleDefault reuseIdentifier:adID];
        }
        for (UIView* view in cell.contentView.subviews) {
            [view removeFromSuperview];
        }
        
        [cell.contentView addSubview:data.webAdView];
        return cell;
    }
}
-(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    //记录点击行
    _indexRow = indexPath.row+1;
    //初始化
    _fam = [DeepleaperFeedAdManager getFeedAdManager];
    _fam.delegate = self;
    _fam.placementID = @"4cc2f95f-629f-446f-83cd-46ca8593f76b";
    _fam.sandBox = YES;
    NSString* urlString = @"http://m.d1cm.com/news/2017032089009.shtml"; //跳转的落地页url
    //请求广告
    [_fam loadAdWithFeedAdType:FEEDAD_WEB andUrl:urlString];
    //跳转落地页
    LandPageViewController* lvc = [LandPageViewController new];
    lvc.url = [NSURL URLWithString:@"http://www.deepleaper.com"];
    [self presentViewController:lvc animated:YES completion:^{
    }];
}

//返回广告成功
-(void)didLoadWebAdView:(DeepleaperWebAdView *)webAdView{
    //加入到数据数组
    FakeData* data = [[FakeData alloc]init];
    data.webAdView = webAdView;
    [_sourceArray insertObject:data atIndex:_indexRow];
    dispatch_async(dispatch_get_main_queue(), ^(){
        //刷新
        [self.tableView reloadData];
    });
    
}
//关闭了广告
-(void)didCloseWebAdView:(DeepleaperWebAdView *)webAdView{
    for (NSInteger i = _sourceArray.count-1 ; i>=0; i--) {
        FakeData* data = _sourceArray[i];
        if (data.webAdView.isClosed) {
            [_sourceArray removeObjectAtIndex:i];
            dispatch_async(dispatch_get_main_queue(), ^{
                [self.tableView deleteRowsAtIndexPaths:@[[NSIndexPath indexPathForRow:i inSection:0]]  withRowAnimation:UITableViewRowAnimationFade];
            });
            break;
        }
    }
}
-(void)didFailLoadWebAdView:(NSError *)error{
    NSLog(@"error:%@",error);
}
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


/*
 #pragma mark - Navigation
 
 // In a storyboard-based application, you will often want to do a little preparation before navigation
 - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
 // Get the new view controller using [segue destinationViewController].
 // Pass the selected object to the new view controller.
 }
 */

@end

File: Example/DeepleaperAdExample_Swift/DeepleaperAdExample_Swift/WebViewController.swift
//
//  WebViewController.swift
//  DeepleaperAdExample_Swift
//
//  Created by 刘畅 on 2017/7/14.
//  Copyright © 2017年 刘畅. All rights reserved.
//

import UIKit
import DeepleaperAd;

class WebViewController: UIViewController,DeepleaperFeedAdManagerDelegate,UITableViewDelegate,UITableViewDataSource  {
    @IBOutlet weak var tableView: UITableView!
    var _sourceArray:[FakeData] = Array();
    var _indexRow = 0;
    var _fam : DeepleaperFeedAdManager!;
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        //初始化数据
        for i in 0..<50{
            let data = FakeData();
            data.title = String.init(format: "第%d行",i);
            _sourceArray.append(data);
        }
    }
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return _sourceArray.count;
    }
    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        return UITableViewAutomaticDimension;
    }
    func tableView(_ tableView: UITableView, estimatedHeightForRowAt indexPath: IndexPath) -> CGFloat {
        return UITableViewAutomaticDimension;
    }
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let data = _sourceArray[indexPath.row];
        if(data.title != nil){
            let ID = "cell";
            var cell = tableView.dequeueReusableCell(withIdentifier: ID);
            if(cell == nil){
                cell = UITableViewCell.init(style: .default, reuseIdentifier: ID);
            }
            cell?.textLabel?.text = data.title;
            return cell!;
        }
        else{
            let adID = "adCell";
            var cell = tableView.dequeueReusableCell(withIdentifier: adID);
            if(cell == nil){
                cell = UITableViewCell.init(style: .default, reuseIdentifier: adID);
            }
            for view in (cell?.contentView.subviews)! {
                view.removeFromSuperview();
            }
            cell?.contentView.addSubview(data.webAdView!);
            return cell!;
        }
    }
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        _indexRow = indexPath.row;
        //初始化
        _fam = DeepleaperFeedAdManager.getFeedAdManager();
        _fam.delegate = self;
        _fam.placementID = "4cc2f95f-629f-446f-83cd-46ca8593f76b";
        _fam.sandBox = true;
        let urlString = "http://m.d1cm.com/news/2017032089009.shtml";
        _fam.loadAd(with: FEEDAD_WEB, andUrl: urlString);
        let lvc = LandPageViewController();
        lvc.url = URL.init(string: urlString);
        self.present(lvc, animated: true) {
        };
    }
    //成功返回
    func didLoad(_ webAdView: DeepleaperWebAdView) {
        //加入到数据源
        let data = FakeData();
        data.webAdView = webAdView;
        _sourceArray.insert(data, at: _indexRow);
        DispatchQueue.main.async {
            self.tableView.reloadData();
        }
    }
    //关闭广告
    func didClose(_ webAdView: DeepleaperWebAdView) {
        for i in 0..<_sourceArray.count{
            let data = _sourceArray[i];
            if(data.nativeAdView?.isClosed == true){
                _sourceArray.remove(at: i);
                DispatchQueue.main.async {
                    self.tableView.deleteRows(at: [IndexPath.init(row: i, section: 0)], with: .fade);
                };
                break;
            }
        }
    }
    //请求失败
    func didFailLoadWebAdView(_ error: Error?) {
        print(error);
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    
    /*
     // MARK: - Navigation
     
     // In a storyboard-based application, you will often want to do a little preparation before navigation
     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
     // Get the new view controller using segue.destinationViewController.
     // Pass the selected object to the new view controller.
     }
     */
    
}
